test/lint/lint-python.py

```diff
@@ -0,0 +1,136 @@
+    'F622,'  # two or more starred expressions in an assignment (a, *b, *c = d)
+    'F631,'  # assertion test is a tuple, which are always True
+    'F632,'  # use ==/!= to compare str, bytes, and int literals
+    'F701,'  # a break statement outside of a while or for loop
+    'F702,'  # a continue statement outside of a while or for loop
+    'F703,'  # a continue statement in a finally block in a loop
+    'F704,'  # a yield or yield from statement outside of a function
+    'F705,'  # a return statement with arguments inside a generator
+    'F706,'  # a return statement outside of a function/method
+    'F707,'  # an except: block as not the last exception handler
+    'F811,'  # redefinition of unused name from line N
+    'F812,'  # list comprehension redefines 'foo' from line N
+    'F821,'  # undefined name 'Foo'
+    'F822,'  # undefined name name in __all__
+    'F823,'  # local variable name … referenced before assignment
+    'F831,'  # duplicate argument name in function definition
+    'F841,'  # local variable 'foo' is assigned to but never used
+    'W191,'  # indentation contains tabs
+    'W291,'  # trailing whitespace
+    'W292,'  # no newline at end of file
+    'W293,'  # blank line contains whitespace
+    'W601,'  # .has_key() is deprecated, use "in"
+    'W602,'  # deprecated form of raising exception
+    'W603,'  # "<>" is deprecated, use "!="
+    'W604,'  # backticks are deprecated, use "repr()"
+    'W605,'  # invalid escape sequence "x"
+    'W606'   # 'async' and 'await' are reserved keywords starting with Python 3.7
+)
+
+
+def check_dependencies():
+    list_out = subprocess.check_output(['pip3', 'list'], stderr=subprocess.DEVNULL).decode('utf-8')
+
+    for dep in DEPS:
+        if dep not in list_out:
+            print(f"Skipping Python linting since {dep} is not installed.")
+            exit(0)
+
+
+def main():
+    check_dependencies()
+
+    if len(sys.argv) > 1:
+        flake8_files = sys.argv[1:]
+    else:
+        files_args = ['git', 'ls-files', '*.py']
+        flake8_files = subprocess.check_output(files_args).decode("utf-8").splitlines()
+
+    flake8_args = ['flake8', '--ignore=B,C,E,F,I,N,W', f'--select={ENABLED}'] + flake8_files
+    flake8_env = os.environ.copy()
+    flake8_env["PYTHONWARNINGS"] = "ignore"
+
+    try:
+        flake8_out = subprocess.run(flake8_args, stdout=subprocess.PIPE, env=flake8_env)
+    except subprocess.CalledProcessError as e:
+        print(e.output.decode("utf-8"), end="")
+        exit(1)
+
+    print(flake8_out.stdout.decode("utf-8"), end="")
+
+    mypy_files = subprocess.check_output(FILES_ARGS).decode("utf-8").splitlines()
+    mypy_args = ['mypy', '--show-error-codes'] + mypy_files
+
+    try:
+        mypy_out = subprocess.check_output(mypy_args, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        print(e.output.decode("utf-8"), end="")
+        exit(1)
```

Comment:
```suggestion
        subprocess.check_call(mypy_args)
    except subprocess.CalledProcessError:
        exit(1)
```

How is this different from the shorter alternative?

See:
* https://docs.python.org/3/library/subprocess.html#subprocess.check_call
* https://github.com/bitcoin/bitcoin/pull/24766#discussion_r843559966

Reply:
Hmm, I tried this initially with `check_call` but in my testing I didn't seem to get the errors to my console. But I guess I did something wrong because with your code it seems to work now. Changed.